components/script/dom/gpurenderpassencoder.rs

```diff
@@ -105,78 +104,99 @@ impl GPURenderPassEncoderMethods for GPURenderPassEncoder {
+                x: *x,
+                y: *y,
+                width: *width,
+                height: *height,
+                min_depth: *min_depth,
+                max_depth: *max_depth,
+            },
+            device_id: self.command_encoder.device_id().0,
+        }) {
+            warn!("Error sending WebGPURequest::RenderPassCommand: {e:?}")
         }
     }
 
     /// <https://gpuweb.github.io/gpuweb/#dom-gpurenderpassencoder-setscissorrect>
     fn SetScissorRect(&self, x: u32, y: u32, width: u32, height: u32) {
-        if let Some(render_pass) = self.render_pass.borrow_mut().as_mut() {
-            wgpu_render::wgpu_render_pass_set_scissor_rect(render_pass, x, y, width, height);
+        if let Err(e) = self.channel.0.send(WebGPURequest::RenderPassCommand {
+            render_pass_id: self.render_pass.0,
+            render_command: RenderCommand::SetScissorRect {
+                x,
+                y,
+                width,
+                height,
+            },
+            device_id: self.command_encoder.device_id().0,
+        }) {
+            warn!("Error sending WebGPURequest::RenderPassCommand: {e:?}")
         }
     }
 
     /// <https://gpuweb.github.io/gpuweb/#dom-gpurenderpassencoder-setblendcolor>
     fn SetBlendConstant(&self, color: GPUColor) {
-        if let Some(render_pass) = self.render_pass.borrow_mut().as_mut() {
-            let colors = match color {
-                GPUColor::GPUColorDict(d) => wgt::Color {
-                    r: *d.r,
-                    g: *d.g,
-                    b: *d.b,
-                    a: *d.a,
-                },
-                GPUColor::DoubleSequence(mut s) => {
-                    if s.len() < 3 {
-                        s.resize(3, Finite::wrap(0.0f64));
-                    }
-                    s.resize(4, Finite::wrap(1.0f64));
-                    wgt::Color {
-                        r: *s[0],
-                        g: *s[1],
-                        b: *s[2],
-                        a: *s[3],
-                    }
-                },
-            };
-            wgpu_render::wgpu_render_pass_set_blend_constant(render_pass, &colors);
+        let color = match color {
+            GPUColor::GPUColorDict(d) => wgt::Color {
+                r: *d.r,
+                g: *d.g,
+                b: *d.b,
+                a: *d.a,
+            },
+            GPUColor::DoubleSequence(mut s) => {
+                if s.len() < 3 {
+                    s.resize(3, Finite::wrap(0.0f64));
+                }
+                s.resize(4, Finite::wrap(1.0f64));
+                wgt::Color {
+                    r: *s[0],
+                    g: *s[1],
+                    b: *s[2],
+                    a: *s[3],
+                }
+            },
+        };
+        if let Err(e) = self.channel.0.send(WebGPURequest::RenderPassCommand {
+            render_pass_id: self.render_pass.0,
+            render_command: RenderCommand::SetBlendConstant(color),
```

Comment:
perhasp add a `send_render_command(&self , cmd: RenderCommand)` method?

Reply:
This is great idea!